src/main/java/core/basesyntax/strategy/impl/BalanceHandlerImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitShopDao;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.utility.ListService;
+import core.basesyntax.utility.impl.ListServiceImpl;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class BalanceHandlerImpl implements OperationHandler {
+    private static List<FruitTransaction> balanceList = new ArrayList<>();
+    private Map<String, Integer> balanceFruitsMap;
+    private ListService listService = new ListServiceImpl();
```

Comment:
Use dependency injection to pass dependencies like listService and balanceFruitsMap instead of instantiating them directly within the class.
This makes the class easier to test and follow the dependency inversion principle.
`private final Map<String, Integer> balanceFruitsMap;
    private final ListService listService;

    public BalanceHandlerImpl(ListService listService) {
        this.listService = listService;
        this.balanceFruitsMap = new HashMap<>();
    }
`